include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -4298,6 +4298,13 @@ struct OrtApi {
   void(ORT_API_CALL* ReleaseROCMProviderOptions)(_Frees_ptr_opt_ OrtROCMProviderOptions* input);
 
   /// @}
+
+  /** \brief Create and register Cuda Allocator
+   *  \param[in] env OrtEnv instance
+   *  \param[in] mem_info OrtMemoryInfo instance
+   */
+  ORT_API2_STATUS(CreateAndRegisterCudaAllocator, _Inout_ OrtEnv* env, _In_ const OrtMemoryInfo* mem_info, _In_ const OrtArenaCfg* arena_cfg,
```

Comment:
I think we should rename this to CreateAndRegisterAllocatorV2 and accept a provider name in addition to other arguments. This function can then potentially support all EPs without hardcoding the name of the EP in the API name.